OneCommunityGlobal/HGNRest
# Jingyi create/edit/delete badges permission

# Description
This PR is related to badge [permission management fixes](https://docs.google.com/spreadsheets/d/1TfiJY9OLDZuyP2UgF0C1wI9tOSLKbS9MiJFH6n1Dsao/edit#gid=0). It enables users with create/edit/delete badges permission to see the changes of badges immediately.

## Related PRS (if any):
This backend PR is related to the development frontend PR.

## Main changes explained:
Whenever the user has the badge management permissions ("createBadges," "updateBadges," or "deleteBadges") and makes changes to badges, the badge management table will fetch the newest data and display the badge immediately. 

## How to test:
1. check into the current branch
2. do `npm install`,  `npm run build`, and `npm start to run the frontend PR locally
4. Clear site data/cache
5. log as admin/owner user
6. go to Other Links -> Permissions Management -> Manage User Permissions -> Badge Permission -> Choose a user without badge management permissions such as 'create badges' 'edit badges' or 'delete badges' -> add the permission for this user -> Scroll down then submit
7. log into the account just granted permission and go to Other Links -> Badge Management to test the permission you just added. For example, create one badge, edit one, or delete one badge.
8. You will see the change immediately on the page.

## Screenshots or videos of changes:
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112589476/d879dabd-4d4e-48d4-a60c-7982e0ceb097
